tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py

```diff
@@ -1319,6 +1319,9 @@ def cost_fn(x):
     def test_state(self, interface, dev, diff_method, grad_on_execution, device_vjp, tol):
         """Test that the state can be returned and differentiated"""
 
+        if "lightning" in getattr(dev, "name", "").lower():
+            pytest.xfail("Lightning does not support state adjoint diff.")
```

Comment:
This is not necessary, I think, because below all tests with `diff_method!="backprop"` are `skip`ped anyways. If we want to switch to `xfail`, I would recommend that we do it for all devices.

Reply:
If `adjoint` diff is requested, the device executes and computes gradients in a single call rather than separate ones. So, this test fails with lightning before it reaches the `pytest.skip` line. The reason why only lightning tests are skipped is because `default.qubit` supports all state-based measurements with adjoint diff now.

Reply:
Oh, I see! So it is not about the differentiation being supported, but about the forward pass with a specific diff_method "in mind" being supported? Thanks! :)